### PR TITLE
filter high level groups and action groups by cluster and index

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
Filters high level groups based on cluster or index panel and also filters the actiongroups based on their associated type

### Category
[Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation]

### Issues Resolved
fix: #1346 
fix: #1347 

### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by cwperks** [comment]
Thank you for the PR @derek-ho ! Would you be able to add a test to verify the new behavior?

**Comment by cwperks** [comment]
@derek-ho Does this fix #1347 too or just #1346? I took a quick glance at the code and it looks like this fixes #1346 , but not #1347

**Comment by DarshitChanpura** [comment]
@derek-ho Would you please fetch the latest changes from main.

File: public/apps/configuration/panels/role-edit/role-edit.tsx [hunk 1]
```diff
@@ -106,12 +106,12 @@ export function RoleEdit(props: RoleEditDeps) {
     }
   }, [addToast, props.action, props.coreStart.http, props.sourceRoleName]);
 
-  const [actionGroups, setActionGroups] = useState<string[]>([]);
+  const [actionGroups, setActionGroups] = useState<Array<[string, ActionGroupItem]>>([]);
```
Comment (RyanL1997) [inline public/apps/configuration/panels/role-edit/role-edit.tsx:109]: Just for my understanding, why are we needing this `ActionGroupItem`, or can we use the original `Action` here?

**Review by RyanL1997 — COMMENTED** [review]
Hi @derek-ho, thanks for taking this on, and it is looking good to me. Just like Craig mentioned above, a testing case would be great.

File: public/apps/configuration/panels/role-edit/role-edit.tsx (hunk 1 above)
Comment (derek-ho) [reply public/apps/configuration/panels/role-edit/role-edit.tsx:109]: If i'm understanding your question correctly I need the whole item here in order to tell whether it is a cluster/index one later on in order to filter them accordingly

File: public/apps/configuration/panels/role-edit/test/role-edit-filtering.test.tsx [hunk 2]
```diff
@@ -0,0 +1,975 @@
...
+          {
+            label: 'cluster_manage_pipelines',
+          },
+        ],
+      },
+      {
+        label: 'Cluster permissions',
+        options: [
+          {
+            label: 'cluster:admin/component_template/delete',
```
Comment (cwperks) [inline public/apps/configuration/panels/role-edit/test/role-edit-filtering.test.tsx:111]: Is it possible to populate this using the value from [CONSTANTS](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/public/apps/configuration/constants.tsx#L33)?

File: public/apps/configuration/panels/role-edit/test/role-edit-filtering.test.tsx [hunk 3]
```diff
@@ -0,0 +1,975 @@
...
+        ],
+      },
+    ]);
+  });
+
+  it('basic index permission panel rendering', async () => {
+    const action = 'create';
+    const buildBreadcrumbs = jest.fn();
+
+    (fetchActionGroups as jest.Mock).mockResolvedValue({
```
Comment (cwperks) [inline public/apps/configuration/panels/role-edit/test/role-edit-filtering.test.tsx:722]: If this is duplicated with the code above, would it be possible to write it once like is done here: https://github.com/opensearch-project/security-dashboards-plugin/pull/1503/files#diff-048f4ea5adf75b37a58ec2e3f1b3e0b3245c0f98e587ec0f8631a0422a7664deR21-R33?

File: public/apps/configuration/panels/role-edit/test/role-edit-filtering.test.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        label: 'Permission groups',
+        options: [
+          {
+            label: 'cluster_manage_pipelines',
+          },
+        ],
+      },
+      {
+        label: 'Cluster permissions',
+        options: CLUSTER_PERMISSIONS.map((x) => {
```
Comment (cwperks) [inline public/apps/configuration/panels/role-edit/test/role-edit-filtering.test.tsx:110]: Much more robust and less lines of code :)

**Review by cwperks — APPROVED** [review]
Looks good to me! Thank you for solving this pesky issue @derek-ho !

**Comment by derek-ho** [comment]
> @derek-ho Does this fix #1347 too or just #1346? I took a quick glance at the code and it looks like this fixes #1346 , but not #1347

This should fix both. It is both filtering action groups (which comes from the network call), and also creates a separate variable, so that only cluster/index respectively permissions from the constants file is included

**Review by scrawfor99 — APPROVED** [review]

**Review by RyanL1997 — APPROVED** [review]
Thanks for the work @derek-ho and I'm approving these changes. However, I think your PR is also impacted by this issue: https://github.com/opensearch-project/security-dashboards-plugin/issues/1506

**Pull request merged**